lib/posix/options/fs.c

```diff
@@ -28,6 +29,9 @@ struct posix_fs_desc {
 	};
 	bool is_dir;
 	bool used;
+#if defined(CONFIG_POSIX_FILE_SYSTEM_FSTAT)
+	char path[MAX_FILE_NAME];
+#endif
```

Comment:
> I'm not sure what the purpose of [`filep`](https://github.com/zephyrproject-rtos/zephyr/blob/7773fe8dd60e9f86760c994090304a0577a1ad2f/include/zephyr/fs/fs_interface.h#L78) is similarly the `dirp` field for directories, but to me that should probably represent a file-system-specific object that could maybe be used as a key.

You are correct regarding the what `dirp` and `filep` things are.
The filep and dirp are pointers to objects returned by FS drivers and VFS does not really understand them, but they have to be unique with respect to FS driver that serves them. There usually is no name to it or a way to associate it to name without storing that info in system. Even using f_open twice  on the same file puts you in risk of having two different filep objects, depending on design of underlying driver.

The VFS is constructed in a way that allows you to reuse one name buffer to open N files and operate on that N objects without remember the name anymore.

We are able to figure out mount points, because VFS stores them to figure out to which backend to talk them. Assuming I am  correct in the above statement we should have unique filep/dirp IDs within scope of mount point so, as you @cfriedt are suggesting that should be possible to match path to ID, but we still need a way to allocate memory for the record with name, somehow.

We could probably solve that with dynamic allocation where POSIX open would allocate common space for name. but that puts us again at MAX_FILE_NAME * number_of_ids cap, but now we are uncertain whether we or something else runs out of memory.

This is hard to solve.

Reply:
@de-nordic - is there a way to look up the parent `filep` or `dirp` for a given node?

If that were the case, and if each `filep` or `dirp` knew its own local name (which I assume would be encoded in the filesystem?) it should be possible to construct a full path with only the runtime overhead of traversing the FS tree, and the runtime allocation of a string of a given length for the full path.

Should be `O(n)` in terms of complexity, where n is the number of nodes in the path.

This would be really sweet.

Reply:
@de-nordic - is there a way to look up the parent `filep` or `dirp` for a given node?

If that were the case, and if each `filep` or `dirp` knew its own local name (which I assume would be encoded in the filesystem?) it should be possible to construct a full path with only the runtime overhead of traversing the FS tree, and the runtime allocation of a string of a given length for the full path.

Should be `O(n)` in terms of complexity, where n is the number of nodes in the path.

This would be really sweet.